nvtabular/loader/backend.py

```diff
@@ -55,16 +74,28 @@ def __init__(
         self.q_out = queue.Queue(qsize)
         self._stop_event = threading.Event()
 
+        # adding for sync behavior
+        self._itr = None
+        self._iterating = False
```

Comment:
Can we just use ```self._itr is None``` as the check to see if we're iterating ?

Reply:
Yeah that actually might make more sense I'll give that a test